### PR TITLE
bugfix/21316-panning-with-overscroll

### LINKED ISSUE #21316: Live data updates prevents panning the chart and affects zoom
#### Expected behaviour
Live data updates do not change behavior for zoom in/out and pan with drag. Chart can be dragged to see past data. Zoom with mouse wheel works consistently.

#### Actual behaviour
When updating bar is within the currently viewed data zoom cycles and pan does not work. Attaching the video with issue shown along with the fiddle (which is basically live data demo). At first half of the video I am trying to drag the chart around, but it is not moving when updating data is within the view. At the second half I am constantly zooming out and zoom just cycles.

#### Live demo with steps to reproduce
[Fiddle](https://jsfiddle.net/stealkvas/p3x5hnyj/1/)

#### Product version
Highstock 11.4.0

#### Affected browser(s)
Tested in Edge, Chrome (latest).

https://github.com/highcharts/highcharts/assets/172513211/5e2e6cce-ff4b-4a01-944c-747fd8ef50a4

### PR DESCRIPTION
**Pull request by kamil-musialowski**
Fixed #21316, panning on axes with overscroll was not possible.

### CODE REVIEW

**Comment by kamil-musialowski** [comment]
Internal explanation: 
Overscroll enables ordinal
Panning triggers `onChartPan()`
Then, this check comes in, which results in running the base `transform()` function https://github.com/highcharts/highcharts/blob/baced054799e4a6490b0dffbe68e0e1d8346869c/ts/Core/Axis/OrdinalAxis.ts#L665
However, since it's an ordinal this check in `transform()` results in false https://github.com/highcharts/highcharts/blob/baced054799e4a6490b0dffbe68e0e1d8346869c/ts/Core/Chart/Chart.ts#L3785
And a setExtremes is not called as a result of that.

Therefore, we also need to check if it's an ordinal, but with evenly spaced data.

File: ts/Core/Chart/Chart.ts [hunk 1]
```diff
@@ -3782,7 +3782,12 @@ class Chart {
             // It is not necessary to calculate extremes on ordinal axis,
             // because they are already calculated, so we don't want to override
             // them.
-            if (!axis.isOrdinal || scale !== 1 || reset) {
+            if (
+                !axis.isOrdinal ||
+                !axis.ordinal?.getExtendedPositions() || // #21316
```
Comment (pawelfus) [inline ts/Core/Chart/Chart.ts:3787]: The comment above says the ticks should be already calculated - could you double-check this? I would rather avoid bloating Core with code related to Stock

**Review by pawelfus — CHANGES_REQUESTED** [review]

File: samples/unit-tests/axis/overscroll/demo.js [hunk 2]
```diff
@@ -74,7 +75,10 @@
             options = getOptions();
```
Comment (DJTechnoo) [inline samples/unit-tests/axis/overscroll/demo.js:75]: Can this line `options = getOptions();`  be removed safely?

**Review by DJTechnoo — APPROVED** [review]
Great! Just one small thing in the comments:

File: ts/Core/Chart/Chart.ts (hunk 1 above)
Comment (kamil-musialowski) [reply ts/Core/Chart/Chart.ts:3787]: Yeah, I agree with this one. How about we just do this?
```suggestion
                axis.options.overscroll || // #21316
```

Tests are passing, I've tested it manually - everything seems to be fine 🤔

File: samples/unit-tests/axis/overscroll/demo.js (hunk 2 above)
Comment (kamil-musialowski) [reply samples/unit-tests/axis/overscroll/demo.js:75]: Wow, good catch!

File: ts/Core/Chart/Chart.ts (hunk 1 above)
Comment (pawelfus) [reply ts/Core/Chart/Chart.ts:3787]: Looks much better. Green light from me 👍

**Review by bm64 — APPROVED** [review]

**Review by pawellysy — APPROVED** [review]
LGTM

**Comment by kamil-musialowski** [comment]
@TorsteinHonsi bump

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**